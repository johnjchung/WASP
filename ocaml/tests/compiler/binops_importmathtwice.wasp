Endpoint intDivideFloatTwice(GET:) {
    int Test = 5 / 2.7;
    int Test2 = 5.5 / 2;
}
