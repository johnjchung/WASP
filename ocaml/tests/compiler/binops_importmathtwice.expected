package main

import (
	"github.com/gin-gonic/gin"
	"math"
	"os"
)

func int_conv(anything interface{}) int {
	switch v := anything.(type) {
		case int:
			return v
		case float64, float32:
			var float_var = v.(float64)
			return int(math.Trunc(float_var))
		default:
			return 0
	}
	return 0
}

func float_conv(anything interface{}) float64 {
 	 switch v := anything.(type) {
		case int:
            return float64(v)
		case float64:
			return v
		default:
			return 0
	}
	return 0.0
}

type intDivideFloatTwice_struct struct {
	Test int `db:"test" json:"test"`
	Test2 int `db:"test2" json:"test2"`
}

func main() {
	router := gin.Default()
	router.GET("/intDivideFloatTwice", Get_intDivideFloatTwice)
	router.Run(":"+os.Getenv("PORT"))
}

func intDivideFloatTwice_struct_wrapper() intDivideFloatTwice_struct {
	return intDivideFloatTwice_struct{Test: int_conv(5) / int_conv(2.7), Test2: int_conv(5.5) / int_conv(2)}
}

func Get_intDivideFloatTwice(c *gin.Context) {
	intDivideFloatTwice_struct := intDivideFloatTwice_struct_wrapper()
	c.JSON(200, intDivideFloatTwice_struct)
}
